### ISSUE TITLE
Implement access code  API for  'tickets/access-codes' route

### ISSUE THREAD

**Issue by sumedh123** [fossasia/open-event-frontend#548]
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

The api has to be implemented at the access codes route.
![api](https://user-images.githubusercontent.com/17925669/28515213-66c17228-707a-11e7-9b6f-5797b61598a3.png)

**Comment by sumedh123** [comment]
Will work on this.

**Comment by niranjan94** [comment]
On hold until ember model table is made compatible with the api. @geekyd
any eta on that?

**Comment by sumedh123** [comment]
@niranjan94 For now, I am just creating model for access-code, will work on the api once the support of the ember data for tables is done. There are two relationships(one one and one many) in the event.js for discount codes? Are we gonna follow up same for access codes too?

**Comment by niranjan94** [comment]
No. Access codes there will only be one relationship. 

Event -> Access Code
One -> Many

**Comment by sumedh123** [comment]
@geekyd @niranjan94 Is there anything different we have to do while rendering data in tables? I am able to get my data in the console but the data is not appearing in my table. Please guide.

**Comment by niranjan94** [comment]
See how it is being done in the `/events` page. Follow the same

**Comment by sumedh123** [comment]
@niranjan94 It is not done yet at '/events'. It is having mock response. Also, I dont see any issue regarding that.

**Comment by niranjan94** [comment]
@sumedh123 maybe you haven't updated your branch. Because, it has been done.

https://github.com/fossasia/open-event-frontend/pull/565

**Comment by sumedh123** [comment]
@geekyd Are the components you created like cell-event.js, cell-button.js generalised? Or specific to the table you have at /events/list

**Comment by sumedh123** [comment]
@niranjan94 As is done for the '/events', for example I have to do for access-codes, do I need to create separate templates for each column of access codes table as is done by geekyd like 'cell-event.hbs', 'cell-date.hbs', cell-button.hbs, cell-tickets.hbs, etc?

**Comment by niranjan94** [comment]
Always re-use stuff. For this you'd need,

`cell-simple` which can be used for any table column with just text. (Use it for column 1, 2, 4, 5)
`cell-label` which can be used for any table column with a label (use for column 6)
`cell-state-action-buttons` with edit, deactivate/active, delete which can be used for discount code table, access code table, users table etc.

**Comment by sumedh123** [comment]
@niranjan94 On hold untill https://github.com/fossasia/open-event-orga-server/issues/4245 resolves.

**Comment by niranjan94** [comment]
@sumedh123 according to https://github.com/fossasia/open-event-orga-server/issues/4245 you have said `/v1/events/{event_identifier}?include=access-codes` gives an error. 

But for this issue, `/v1/events/{event_identifier}/access-codes`, is only required.

**Comment by sumedh123** [comment]
@niranjan94 Considering the above URL:
I am not able to get the access codes using:
```
this.get('store').findRecord('event', this.paramsFor('events.view').event_id).get('accessCodes');
```
Although I can see the relationship is available between event and access-codes in their individual responses queried manually. That's where I am stuck.

**Comment by niranjan94** [comment]
@sumedh123 does

```
this.modelFor('events.view').get('accessCodes')
``` 

work ? (after adding the model and relationship of course)

**Comment by niranjan94** [comment]
And please give more info than "I am not able to get the access codes". Something like a code sample or a stacktrace would be really useful to assist you.

**Comment by sumedh123** [comment]
@niranjan94 ```this.modelFor('events.view').get('accessCodes')``` works.

**Comment by niranjan94** [comment]
@sumedh123 you're previous code too is partly correct. But the reason it did not work is, you tried to get access code from a promise.

`findRecord` always returns a promise. So the following would have probably worked. 

```js
this.get('store').findRecord('event', this.paramsFor('events.view').event_id)
    .then(event => {
        event.get('accessCodes')
    });
```

But using `modelFor` is a more shorter way to do this :)
